File: core/src/com/alta_v2/mediator/screen/context/ContextFactory.java
package com.alta_v2.mediator.screen.context;

import com.alta_v2.model.MenuDefinitionModel;
import com.alta_v2.model.TiledMapDefinitionModel;

public interface ContextFactory {

    /**
     * Creates the context for menu screen.
     */
    ScreenContext createMenuContext(MenuDefinitionModel definition);

    /**
     * Creates the context for tiled map screen.
     */
    ScreenContext createTiledMapContext(TiledMapDefinitionModel definition);

}

File: core/src/com/alta_v2/mediator/screen/context/ContextFactory.kt
package com.alta_v2.mediator.screen.context

import com.alta_v2.model.MenuDefinitionModel
import com.alta_v2.model.TiledMapDefinitionModel

interface ContextFactory {
    /**
     * Creates the context for menu screen.
     */
    fun createMenuContext(definition: MenuDefinitionModel): ScreenContext

    /**
     * Creates the context for tiled map screen.
     */
    fun createTiledMapContext(definition: TiledMapDefinitionModel): ScreenContext
}